IonicPortals/src/main/kotlin/io/ionic/portals/PortalsPlugin.kt

```diff
@@ -1,45 +1,50 @@
 package io.ionic.portals
 
+import android.util.Log
 import com.getcapacitor.*
 import com.getcapacitor.annotation.CapacitorPlugin
 import org.json.JSONException
 import org.json.JSONObject
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicInteger
 
-@CapacitorPlugin(name = "Portals")
-class PortalsPlugin : Plugin() {
+class PortalsPubSub {
+    private var subscriptions = ConcurrentHashMap<String, MutableMap<Int, (data: SubscriptionResult) -> Unit>>()
+    private var subscriptionRef = AtomicInteger(0)
 
-    companion object {
-        @JvmStatic
-        var subscriptions = mutableMapOf<String, MutableMap<Int, (data: SubscriptionResult) -> Unit>>()
-        @JvmStatic
-        var subscriptionRef = 0
-        @JvmStatic
-        fun publish(topic: String, data: Any?) {
-            subscriptions[topic]?.let {
-                for ((ref, listener) in it) {
-                    val result = SubscriptionResult(topic, data, ref)
-                    listener(result)
-                }
+    fun publish(topic: String, data: Any?) {
+        subscriptions[topic]?.let {
+            for ((ref, listener) in it) {
+                val result = SubscriptionResult(topic, data, ref)
+                listener(result)
             }
         }
+    }
 
-        @JvmStatic
-        fun subscribe(topic: String, callback: (result: SubscriptionResult) -> Unit): Int {
-            subscriptionRef++
-            subscriptions[topic]?.let { subscription ->
-                subscription[subscriptionRef] = callback
-            } ?: run {
-                val subscription = mutableMapOf(subscriptionRef to callback)
-                subscriptions[topic] = subscription
-            }
-            return subscriptionRef
+    fun subscribe(topic: String, callback: (result: SubscriptionResult) -> Unit): Int {
+        val ref = subscriptionRef.incrementAndGet()
+        subscriptions[topic]?.let { subscription ->
+            subscription[ref] = callback
+        } ?: run {
+            val subscription = mutableMapOf(ref to callback)
+            subscriptions[topic] = subscription
         }
+        return ref
+    }
+
+    fun unsubscribe(topic: String, subscriptionRef: Int) {
+        subscriptions[topic]?.remove(subscriptionRef)
+    }
 
+    companion object {
         @JvmStatic
-        fun unsubscribe(topic: String, subscriptionRef: Int) {
-            subscriptions[topic]?.remove(subscriptionRef)
-        }
+        val shared = PortalsPubSub()
```

Comment:
I don't really know if this is a very common convention in Android/Kotlin/JVM stuff when providing a base singleton to use.

Reply:
I think so